python_modules/dagster/dagster/_config/structured_config.py

```diff
@@ -216,20 +216,32 @@ def create_io_manager_to_pass_to_user_code(self, context) -> IOManager:
         return self
 
 
+WRAPPER_TYPES: Dict[int, Callable[[ConfigType], ConfigType]] = {
+    SHAPE_SINGLETON: lambda x: x,
+    SHAPE_LIST: Array,
+}
```

Comment:
It would be helpful to wrap this accessing this in a function and then explaining what is happening here. Questions I have:

1) Why is the key an `int`? I can infer that pydantic assigns int ids to things, but I'd rather see that laid out specifically
2) Why is this called "wrapped"? Doesn't seem like anything is being wrapped, but mapped. Let's work on a better name.

Reply:
Moved to a function instead of indexing into this map. Aliased the `int` type to something more descriptive as well.

Hopefully the `wrapped` name makes a bit more sense with the docstring, it's because it's wrapping the inferred config type in whichever data structure is inferred from the Pydantic shape (wrapped in a list, map, etc).